### PR TITLE
Initial wiring of encryption through the command package

### PR DESCRIPTION
**Pull request by cam72cam**
This PR fully wires the state encryption package into the command structure.

TODO:
* [x] Make sure Backend/StateFile are used in the correct locations
* [x] Take a closer look at changes made to the provider interface for encrypted remote state access
* [ ] Add a boat load of tests

Related to #1030 

## Target Release

1.7.0

### CODE REVIEW

File: internal/backend/local/backend.go [hunk 1]
```diff
@@ -106,9 +106,6 @@ func New(enc encryption.StateEncryption) *Local {
 // NewWithBackend returns a new local backend initialized with a
 // dedicated backend for non-enhanced behavior.
 func NewWithBackend(backend backend.Backend, enc encryption.StateEncryption) *Local {
-	if backend == nil && enc == nil {
```
Comment (cam72cam) [inline internal/backend/local/backend.go:109]: This is actually valid for some commands that do not use encryption, like get schema

File: internal/grpcwrap/provider6.go [hunk 2]
```diff
@@ -373,6 +374,7 @@ func (p *provider6) ReadDataSource(_ context.Context, req *tfplugin6.ReadDataSou
 		TypeName:     req.TypeName,
 		Config:       configVal,
 		ProviderMeta: metaVal,
+		Encryption:   encryption.StateEncryptionDisabled(), // Not supported via GRCP
```
[the remark this answers is not in the document]
Comment (cam72cam) [reply internal/grpcwrap/provider6.go:377]: Thanks for catching that!

File: internal/command/meta_encryption.go [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+		panic(err)
+	}
+	if err := reg.RegisterMethod(aesgcm.New()); err != nil {
+		panic(err)
+	}
+
+	cfg := module.Encryption
+	var diags tfdiags.Diagnostics
+
+	envName := "TF_ENCRYPTION"
```
Comment (Yantrio) [inline internal/command/meta_encryption.go:49]: Make this a constant with a comment explaining what it does. Also please add this to the docs here: https://opentofu.org/docs/cli/config/environment-variables

File: internal/command/state_meta.go [hunk 4]
```diff
@@ -28,17 +28,17 @@ type StateMeta struct {
...
-func (c *StateMeta) State() (statemgr.Full, error) {
+func (c *StateMeta) State(enc encryption.Encryption) (statemgr.Full, error) {
 	var realState statemgr.Full
 	backupPath := c.backupPath
 	stateOutPath := c.statePath
 
 	// use the specified state
 	if c.statePath != "" {
-		realState = statemgr.NewFilesystem(c.statePath, encryption.StateEncryptionTODO())
+		realState = statemgr.NewFilesystem(c.statePath, encryption.StateEncryptionDisabled()) // User specified state file should not be encrypted
```
Comment (Yantrio) [inline internal/command/state_meta.go:38]: We should check manually that this works with something like `tofu output`

File: internal/encryption/base.go [hunk 5]
```diff
@@ -100,16 +111,20 @@ func (s *baseEncryption) decrypt(data []byte, validator func([]byte) error) ([]b
...
-		if err != nil {
+		verr := validator(data)
+		if verr != nil {
 			// Nope, just bad input
-			return nil, fmt.Errorf("unable to determine data structure during decryption: %w", err)
+
+			// Return the outer json error if we have one
+			if err != nil {
+				return nil, fmt.Errorf("invalid data format for decryption: %w, %w", err, verr)
+			}
```
[the remark this answers is not in the document]
Comment (cam72cam) [reply internal/encryption/base.go:124]: 002b5d5c7f

**Pull request merged**